src/legacy/core_plugins/data/public/setup.ts

```diff
@@ -17,11 +17,21 @@
  * under the License.
  */
 
+import { npSetup } from 'ui/new_platform';
 import { DataPlugin } from './index';
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const { IndexPatterns } = require('ui/index_patterns/index');
 
 /**
  * We export data here so that users importing from 'plugins/data'
  * will automatically receive the response value of the `setup` contract, mimicking
  * the data that will eventually be injected by the new platform.
  */
-export const data = new DataPlugin().setup();
+
+const setupPlugins = {
+  ...npSetup.plugins,
```

Comment:
@streamich if `npSetup.plugins` houses plugins that are officially moved to the new platform, and we are shimming our plugins so that others can import from the legacy world and still receive the contracts in the "shape" of the new platform, then the `__LEGACY` convention here is primarily only useful for us (App Arch) to track stuff that is being imported from `ui/public`.

What I mean by this is that the majority of Kibana users who would be importing from `ui/*` will now actually be importing from one of our plugins -- data, visualizations, embeddables, and a few others. And those are already being built to mirror what will eventually be included in `npSetup.plugins`, so it therefore wouldn't belong in `__LEGACY`

All this is to say: this pattern makes sense for us, but the more I think about it, for most Kibana devs the solution may end up looking more like this in practice:

```javascript
import { setup as visSetup } from '../core_plugins/visualizations/public/legacy';
const setupPlugins = {
  ...npSetup.plugins,
  visualizations: visSetup,
  __LEGACY: {
    // probably just a few miscellaneous things that aren't yet in a shim or in the NP
  }
};
```

Any thoughts on this?

cc @lizozom @ppisljar

Reply:
yes, i completely agree, this will mostly serve us, and we might recommend this approach to other teams with things we won't be able to move in time.